src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1495,6 +1499,40 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
         return false;
     }
 
+    /**
+     * Check if a tag has been requested already or is in the inventory. If not in the inventory and not requested already, create request
+     *
+     * @param tag the requested tag.
+     * @return true if in the inventory, else false.
+     */
+    public boolean checkIfRequestForTagExistOrCreateAsynch(@NotNull final Tag<Item> tag, final int count, final String description)
+    {
+        if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), s -> s.getItem().isIn(tag) && s.getCount() >= count))
+        {
+            return true;
+        }
+
+        if (InventoryUtils.getItemCountInProvider(getOwnBuilding(),
+                itemStack -> itemStack.getItem().isIn(tag)) >= count &&
+                InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                        getOwnBuilding(), itemStack -> itemStack.getItem().isIn(tag),
+                        count,
+                        worker.getInventoryCitizen()))
+        {
+            return true;
+        }
+
+        final List<ItemStack> stacks = tag.getAllElements().stream().map(itemIn -> new ItemStack(itemIn, count)).collect(Collectors.toList());
```

Comment:
That is quite the creative way to make a tag based request. Although I'm unsure we shouldn't create just a TagBased request type new instead of doing it the hacky way.